odoo/addons/base/models/ir_rule.py

```diff
@@ -141,21 +149,36 @@ def _compute_domain(self, model_name, mode="read"):
         # browse user and rules as SUPERUSER_ID to avoid access errors!
         eval_context = self._eval_context()
         user_groups = self.env.user.groups_id
-        global_domains = []                     # list of domains
-        group_domains = []                      # list of domains
-        for rule in rules.sudo():
-            # evaluate the domain for the current user
-            dom = safe_eval(rule.domain_force, eval_context) if rule.domain_force else []
-            dom = expression.normalize_domain(dom)
-            if not rule.groups:
-                global_domains.append(dom)
-            elif rule.groups & user_groups:
-                group_domains.append(dom)
-
-        # combine global domains and group domains
+        global_domains = []  # list of lists of domains grouped by global_grouping_criteria
+        group_domains = []  # list of domains
+        for criteria, grouped_rules in rules.sudo().groupby(
+            lambda r: str(r.global_grouping_criteria)
+        ):
+            criteria = False if criteria == "False" else criteria  # FIXME: Don't str() in the groupby() somehow!
```

Comment:
Why do we need the str? Groupyby should cope with grouping by boolean false value.

Reply:
It didn't work without str() when i tried, got an error